client/components/books-list.jsx

```diff
@@ -1,27 +1,40 @@
 import React from 'react';
 
 function Book(props) {
-  return (
-    <li className='flex margin-top'>
-      <img src={props.book.book_image} alt={props.book.title}/>
-      <div className='content-holder'>
-        <p className='number-of-weeks'>{props.book.weeks_on_list} WEEKS ON THE LIST</p>
-        <h3 className='title margin-top '>{props.book.title}</h3>
-        <p className='author margin-top '>by {props.book.author}</p>
-        <p className='description margin-top '>{props.book.description}</p>
-      </div>
+  const imgageUrl = props.book.book_image;
+  const numberWeeks = props.book.weeks_on_list;
+  if (props.book.isbns[0]) {
+    const isbn = props.book.isbns[0].isbn10;
+    return (
+      <a
+        href={`#details?isbn=${isbn}&imageurl=${imgageUrl}&n=${numberWeeks}`}
+      className='flex margin-top a-book'>
+        <li className='flex margin-top'>
+          <img src={imgageUrl} alt={props.book.title} />
+          <div className='content-holder'>
+            <p className='number-of-weeks '>{props.book.weeks_on_list} WEEKS ON THE LIST</p>
+            <p className='title margin-top'>{props.book.title}</p>
+            <p className='author margin-top'>by {props.book.author}</p>
+            <p className='description margin-top'>{props.book.description}</p>
+            <p href={`#details?isbn=${isbn}`} className='more-details-p'>More Details...</p>
+          </div>
+        </li>
+      </a>
 
-    </li>
-  );
+    );
+  } else {
+    delete props.book.isbns[0];
+  }
 }
 
 export default function BookList(props) {
   return (
     <ul>
       {
-
         props.books.map(book => {
-          return <Book key={book.title} book={book} />;
+          return (
+            <Book key={book.title} book={book} />
+          );
         })
       }
```

Comment:
Instead of this `if (props.book.isbns[0]) {`, use `.filter()` before `.map` to only map books with the isbn.

```jsx
{
  props.books
    .filter(book => book.isbns.length > 0)
    .map(book => <Book key={book.title} book={book} />)
}
```

Reply:
<img width="542" alt="Screen Shot 2022-09-15 at 9 20 55 AM" src="https://user-images.githubusercontent.com/106217355/190456266-b254659d-612c-443e-a372-9d9aa8632bda.png">